python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -546,23 +550,120 @@ def get_materialization_status_for_asset_health(self, graphene_info: ResolveInfo
+                    AssetCheckExecutionResolvedStatus.SKIPPED,
+                ]:
+                    # the last check is still in progress or is skipped, so we want to check the status of
+                    # the latest completed check instead
+                    if summary_record.last_completed_check_execution_record is None:
+                        # the check hasn't been executed prior to this in progress check
+                        all_checks_executed = False
+                        continue
+                    last_check_execution_status = (
+                        await summary_record.last_completed_check_execution_record.resolve_status(
+                            graphene_info.context
+                        )
+                    )
+                    last_check_evaluation = (
+                        summary_record.last_completed_check_execution_record.evaluation
+                    )
+
+                check_statuses.append(last_check_execution_status)
+                if last_check_execution_status == AssetCheckExecutionResolvedStatus.FAILED:
+                    # failed checks should always have an evaluation, but default to ERROR if not
+                    check_failure_severities.append(
+                        last_check_evaluation.severity
+                        if last_check_evaluation
+                        else AssetCheckSeverity.ERROR
+                    )
+                if (
+                    last_check_execution_status
+                    == AssetCheckExecutionResolvedStatus.EXECUTION_FAILED
+                ):
+                    # EXECUTION_FAILED checks may not have an evaluation, and we want to show these as
+                    # degraded health anyway.
+                    check_failure_severities.append(AssetCheckSeverity.ERROR)
+
+            if len(check_statuses) == 0:
+                # checks have never been executed
+                return GrapheneAssetHealthStatus.UNKNOWN
+
+            if any(
+                status == AssetCheckExecutionResolvedStatus.FAILED
+                or status == AssetCheckExecutionResolvedStatus.EXECUTION_FAILED
+                for status in check_statuses
+            ):
+                if all(
+                    severity == AssetCheckSeverity.WARN for severity in check_failure_severities
+                ):
+                    # all failed checks are with warning severity
+                    return GrapheneAssetHealthStatus.WARNING
+                # at least one failing check had error severity
+                return GrapheneAssetHealthStatus.DEGRADED
+            # since we are only looking at the latest completed execution for each check, if there
+            # are no failed checks, then all checks must have succeeded
+            if not all_checks_executed:
+                # all executed checks succeeded, but some checks have never been executed, so we consider this a warning
```

Comment:
I disagree with this case being a warning.

In the state that you've executed no checks, you have an unknown state. If you then execute one of the checks successfully, it doesn't make sense to me to switch to a higher severity "warning" state compared to the prior "unknown" state. I think this should be a healthy state if any check is healthy and the others are all unknown.